modules/developer_guide.ipynb

```diff
@@ -0,0 +1,812 @@
+     "text": [
+      "2023-02-21 12:29:15,364 - INFO - Verified 'Task05_Prostate.tar', md5: 35138f08b1efaef89d7424d2bcc928db.\n",
+      "2023-02-21 12:29:15,365 - INFO - File exists: Task05_Prostate.tar, skipped downloading.\n",
+      "2023-02-21 12:29:15,366 - INFO - Non-empty folder exists in Task05_Prostate, skipped extracting.\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Loading dataset: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 26/26 [00:00<00:00, 99955.92it/s]"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "number of subjects: 26\n",
+      "The first element in the dataset is\n",
+      "{'image': 'Task05_Prostate/imagesTr/prostate_46.nii.gz', 'label': 'Task05_Prostate/labelsTr/prostate_46.nii.gz'}.\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataset = monai.apps.DecathlonDataset(\n",
+    "    root_dir=\"./\", task=\"Task05_Prostate\", section=\"training\", transform=None, download=True\n",
+    ")\n",
+    "print(f\"\\nnumber of subjects: {len(dataset)}\")\n",
+    "print(f\"The first element in the dataset is\\n{dataset[0]}.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "POwC6_uvLFtt"
+   },
+   "source": [
+    "The code section has created a `DecathlonDataset` object by downloading \"Task05_Prostate.tar\" (`download=True`) from the public repository, unzipping it, and parsing the JSON file provided by the archive.\n",
+    "\n",
+    "This is a three-class segmentation task for delineating prostate transitional zone and peripheral zone (background, TZ, PZ classes).  The input has two modalities, that is, T2 and ADC MRI."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "3yRel2e3hxLo"
+   },
+   "source": [
+    "`len(dataset)` and `dataset[0]` queries the size of the dataset, and fetches the first element of the dataset respectively.  As we haven't specified any image transforms for the dataset, the outputs of this iterable dataset are just pairs of image and segmentation file names."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "_aMAATwVKmJy"
+   },
+   "source": [
+    "## Flexible image data transformations\n",
+    "This section introduces MONAI transforms that convert the file names into data arrays in memory, which will be ready to be consumed by the deep learning models. More examples on the pre-processing pipelines are available in the [tutorial repository](https://github.com/Project-MONAI/tutorials), as well as [the developer's guide](https://github.com/Project-MONAI/MONAI/wiki/Developer-Guide-Transforms). Here we briefly cover the main features for image pre-processing."
```

Comment:
Hi @wyli , thank you for putting this together! This notebook receives the best feedback from the VDR process, as it is clear and the contents are helpful. So I think it is important beginning for the new MONAI users.

Here in this paragraph, we point users to the tutorial repo to find transforms example. I think it is not an easy thing for new users.
I would list some notebooks, as the one I found below. Also, I would use relative paths instead of web link to ensure they can open it in Google Colab and Jupyter Lab, and other platforms.
```
- [3d_image_transforms](3d_image_transforms.ipynb)
- [integrate_3rd_party_transforms](integrate_3rd_party_transforms.ipynb)
- [transforms_update_meta_data](transforms_update_meta_data.ipynb)
```